upsetplot/data.py

```diff
@@ -57,20 +67,30 @@ def generate_counts(seed=0, n_samples=10000, n_categories=3):
         Number of samples to generate statistics over
     n_categories : int
         Number of categories (named "cat0", "cat1", ...) to generate
+    extra_columns: int
+        Number of additional features to be use to generate each
+        sample (value, value1, value2, ...)
 
     Returns
     -------
-    Series
-        Counts indexed by boolean indicator mask for each category.
+    Series or DataFrame
+        A Series of counts indexed by boolean indicator mask for each category,
+        when ``extra_columns`` is 0. Otherwise a DataFrame with column ``value``
+        equivalent to the value produced when ``extra_columns`` is 0, as well as
+        further random variables ``value1``, ``value2``, for extra columns.
 
     See Also
     --------
     generate_samples : Generates a DataFrame of samples that these counts are
         derived from.
     """
+    assert extra_columns >= 0, 'extra_columns parameter should be possitive'
     df = generate_samples(seed=seed, n_samples=n_samples,
-                          n_categories=n_categories)
-    return df.value.groupby(level=list(range(n_categories))).count()
+                          n_categories=n_categories,
+                          extra_columns=extra_columns)
+    df.drop('index', axis=1, inplace=True)
+    df = df if extra_columns > 0 else df.value
+    return df.groupby(level=list(range(n_categories))).count()
```

Comment:
I don't think counting is meaningful for the extra columns. Maybe we should use a different aggregate?

Reply:
Or maybe we shouldn't offer this functionality in `generate_counts`, making things somewhat simpler.